src/System.Drawing.Primitives/tests/ColorTests.netcoreapp.cs

```diff
@@ -0,0 +1,190 @@
+
+        [Theory, MemberData(nameof(AllKnownColors))]
+        public void ToKnownColorMatchesButIsNotKnown(KnownColor known)
+        {
+            Color color = Color.FromKnownColor(known);
+            Color match = Color.FromArgb(color.A, color.R, color.G, color.B);
+            Assert.Equal((KnownColor)0, match.ToKnownColor());
+        }
+
+        [Fact]
+        public void FromOutOfRangeKnownColorToKnownColor()
+        {
+            Color color = Color.FromKnownColor((KnownColor)(-1));
+            Assert.Equal((KnownColor)0, color.ToKnownColor());
+
+            color = Color.FromKnownColor(0);
+            Assert.Equal((KnownColor)0, color.ToKnownColor());
+
+            color = Color.FromKnownColor(KnownColor.MenuHighlight + 1);
+            Assert.Equal((KnownColor)0, color.ToKnownColor());
+        }
+
+        [Theory, MemberData(nameof(SystemColors))]
+        public void IsSystemColorTrue(KnownColor known) => Assert.True(Color.FromKnownColor(known).IsSystemColor);
+
+        [Theory, MemberData(nameof(NonSystemColors))]
+        public void IsSystemColorFalse(KnownColor known) => Assert.False(Color.FromKnownColor(known).IsSystemColor);
+
+        [Theory, MemberData(nameof(SystemColors))]
+        public void IsSystemColorFalseOnMatching(KnownColor known)
+        {
+            Color color = Color.FromKnownColor(known);
+            Color match = Color.FromArgb(color.A, color.R, color.G, color.B);
+            Assert.False(match.IsSystemColor);
+        }
+
+        [Fact]
+        public void IsSystemColorOutOfRangeKnown()
+        {
+            Color color = Color.FromKnownColor((KnownColor)(-1));
+            Assert.False(color.IsSystemColor);
+
+            color = Color.FromKnownColor(0);
+            Assert.False(color.IsSystemColor);
+
+            color = Color.FromKnownColor(KnownColor.MenuHighlight + 1);
+            Assert.False(color.IsSystemColor);
+        }
+
+        [Theory, MemberData(nameof(AllKnownColors))]
+        public void IsKnownColorTrue(KnownColor known)
+        {
+            Assert.True(Color.FromKnownColor(known).IsKnownColor);
+        }
+
+        [Theory, MemberData(nameof(AllKnownColors))]
+        public void IsKnownColorMatchFalse(KnownColor known)
+        {
+            Color color = Color.FromKnownColor(known);
+            Color match = Color.FromArgb(color.A, color.R, color.G, color.B);
+            Assert.False(match.IsKnownColor);
+        }
+
+        [Fact]
+        public void IsKnownColorOutOfRangeKnown()
+        {
+            Color color = Color.FromKnownColor((KnownColor)(-1));
+            Assert.False(color.IsKnownColor);
+
+            color = Color.FromKnownColor(0);
+            Assert.False(color.IsKnownColor);
+
+            color = Color.FromKnownColor(KnownColor.MenuHighlight + 1);
+            Assert.False(color.IsKnownColor);
+        }
+
+        [Fact]
+        public void GetHashCodeForUnknownNamed(string name1, string name2)
```

Comment:
I said the thing no hacker should ever say to themselves. I said, "oh, that'll just be a quick change here".